solver/src/settlement.rs

```diff
@@ -139,35 +180,47 @@ impl Settlement {
 // The difference between what you were willing to sell (executed_amount * limit_price) converted into reference token (multiplied by buy_token_price)
 // and what you had to sell denominated in the reference token (executed_amount * buy_token_price)
 fn buy_order_surplus(
-    sell_token_price: U256,
-    buy_token_price: U256,
-    sell_amount_limit: U256,
-    buy_amount_limit: U256,
-    executed_amount: U256,
-) -> Option<U256> {
-    executed_amount
+    sell_token_price: &BigRational,
+    buy_token_price: &BigRational,
+    sell_amount_limit: &BigRational,
+    buy_amount_limit: &BigRational,
+    executed_amount: &BigRational,
+) -> Option<BigRational> {
+    let res = executed_amount
         .checked_mul(sell_amount_limit)?
         .checked_div(buy_amount_limit)?
         .checked_mul(sell_token_price)?
-        .checked_sub(executed_amount.checked_mul(buy_token_price)?)
+        .checked_sub(&executed_amount.checked_mul(buy_token_price)?)?;
+    // Should we simply return 0 when the order fails to satisfy the limit price,
```

Comment:
I think it's fine to treat this the same way as an overflow in the computation. We should probably log an error so we are being made aware of it.

Reply:
Done!

Reply:
Done here: 119b062